pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -280,6 +283,29 @@ func (m *Multicluster) ClusterAdded(cluster *multicluster.Cluster, clusterStopCh
 	return nil
 }
 
+// checkCanLead returns true if the caller has sufficient authority to become elected leader.
+func (m *Multicluster) checkCanLead(client kubelib.Client) bool {
+	if !features.ExternalIstiod {
+		return false
+	}
+	s := &authorizationapi.SelfSubjectAccessReview{
+		Spec: authorizationapi.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationapi.ResourceAttributes{
+				Namespace: m.opts.SystemNamespace,
+				Verb:      "create",
+				Group:     "",
+				Resource:  "configmaps",
+			},
+		},
+	}
+	response, err := client.Kube().AuthorizationV1().SelfSubjectAccessReviews().Create(context.TODO(), s, metav1.CreateOptions{})
+	if err != nil {
+		log.Errorf("could not determine leadership permission: %v", err)
```

Comment:
This means a transient error will permanently change behavior. IMO this is pretty dangerous

Reply:
Could return true in this case, which would then result in the same permission errors in the log that this PR is trying to suppress, but at least we've suppressed them most of the time.